fileutil/file.go

```diff
@@ -13,7 +13,7 @@ import (
 )
 
 var (
-	shebangRe = regexp.MustCompile(`^#!\s?/(usr/)?bin/(env\s+)?(sh|bash|mksh|bats|zsh)\s`)
+	shebangRe = regexp.MustCompile(`^#!\s?/(usr/)?bin/(env\s+)?(sh|bash|mksh|bats|zsh)\s?`)
```

Comment:
The `\s` there is for good reason; see 31883e5e05b63c0a5a8f97e90d3f0576fbc86fec. Why add the `?` at the end? Note that we don't have a trailing `$` either.

Reply:
Ah, right. The regular expression was only used on the unparsed file bytes before, so the comment would pretty much always be followed by a newline - which is also whitespace. Since you're using it on a parsed comment now, which omits the newline, then there's no whitespace.

I think you want `(\s|$)` then - because otherwise `\s?` will mean we match `#!/usr/bin/shoo`, as `sh`, whereas we only want to match `#!/usr/bin/sh some args` or `#!/usr/bin/sh`.